### PR TITLE
channelcommands.js crushed some bugs and jslinted

### PR DESCRIPTION
**Pull request by NightfallAlicorn**
- Lots of script errors that printed in the server console window are now no longer.
- Tidied with JSLint and improved written code.
- Added missing command in help `/topicparts`.
- Moved `/topic` in help to display for all users since it is accessible to all channel. Including note that only channel mods or higher can change it.

### CODE REVIEW

**Comment by NightfallAlicorn** [comment]
Good dog. *throws Houndy a bone*

File: scripts/channelcommands.js [hunk 1]
```diff
@@ -217,11 +220,13 @@ exports.handleCommand = function (src, command, commandData, tar, channel) {
             channelbot.sendMessage(src, "Correct usage is /" + command + " [number]. Separate multiple part numbers with a space.", channel);
             return;
         }
-        var parts = commandData.indexOf(":") !== -1 ? commandData.split(":") : commandData.split(" ");
-        var topic = SESSION.channels(channel).topic, duplicates = [];
+        var i,
```
Comment (Fuzzysqurl) [inline scripts/channelcommands.js]: this annoys me :[

File: scripts/channelcommands.js [hunk 2]
```diff
@@ -231,33 +234,33 @@ exports.handleCommand = function (src, command, commandData, tar, channel) {
...
         return;
-    }   
+    }
     if (command === "topicparts") {
         var topic = SESSION.channels(channel).topic;
-        if (typeof topic == "string" && topic !== "") {
+        if (typeof topic === "string" && topic !== "") {
             var i = 0;
-            topic = topic.split(" | ").map(function(part) { 
+            topic = topic.split(" | ").map(function (part) {
```
Comment (PokeWorldBW) [inline scripts/channelcommands.js]: This could probably use Config.topic_delimiter instead of " | "  (same with line 257) =D

**Comment by NightfallAlicorn** [comment]
The only thing I'm concerned about is that it's using lowercase and underscore instead of camelcase. If you recommend changing it, I do it if you want?

**Comment by PokeWorldBW** [comment]
Eh, it's not really important. I just thought that you might want to be consistent since Config.topic_delimiter appears in other places instead of " | ", and, if for some reason the delimiter were changed in Config, this would not properly change with it.

**Pull request merged**